java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -862,6 +862,13 @@ public void setBytes(long index, ByteBuffer src) {
         src.position(src.position() + length);
       } else {
         // copy word at a time
+        while (length - 128 >= LONG_SIZE) {
+          for (int x = 0; x < 16; x++) {
+            PlatformDependent.putLong(dstAddress, src.getLong());
+            length -= LONG_SIZE;
+            dstAddress += LONG_SIZE;
+          }
+        }
```

Comment:
awesome!

Reply:
This is not a comment. I am just curious:
The biggest register is 512 bits wide for AVX-512. So x should be within 512 / 8 = 64?

Reply:
I will admit to not being particularly principled in this selection.  I don't think this is solely about SIMD.  Having a fixed size loop allows for unrolling which even without SIMD avoids branches.  I tried changing this just to the size of AVX registers on my box and saw a small slowdown on the BitVectorHelper tests, this could have been noise.  I don't believe unrolling was happening here before because based on my reading it seems that stride lengths of 8 are not considered for unrolling.

Reply:
I see. Thanks for the explanation.